lib/internal/validators.js

```diff
@@ -257,6 +257,20 @@ const validateObject = hideStackFrames(
     }
   });
 
+/**
+ * @callback validateDictionary
+ * @param {*} value
+ * @param {string} name
+ */
+
+/** @type {validateDictionary} */
+const validateDictionary = hideStackFrames(
```

Comment:
Why is it called `validateDictionary` if functions are also allowed?

Reply:
That's the name the WHATWG spec is using, see https://webidl.spec.whatwg.org/#es-dictionary

Reply:
Can you add the link to the whatwg spec?

Reply:
Specifically for this line `validateDictionary`. It would be good to know why we use the keyword `dictionary`.

Reply:
I see. In this case it seems we could also just do `if (value !== undefined && typeof value !== 'object' && typeof value !== 'function')`, or save `const type = typeof value` and check for `if (type !== 'undefined' && type !== 'object' && type !== 'function')`. It does not seem necessary to say `Object or Function` in the error message if we define that functions are also objects (that implement `[[Call]]`). `undefined, null or JavaScript objects`, as how the spec describes them, should be enough.

Reply:
Regarding the error message, here how it's handled on other runtimes:

- Firefox: `TypeError: Blob constructor: Argument 2 can't be converted to a dictionary.`
- Safari: `TypeError: Type error`
- Chromium: `TypeError: Failed to construct 'Blob': The provided value is not of type 'BlobPropertyBag'.`
- Deno: `TypeError: Failed to construct 'Blob': Argument 2 can not be converted to a dictionary`

Maybe we should use the term  `dictionary`? Otherwise it might be OK to just keep `Object`? Functions do inherit from `Object` after all, so it would be technically correct, and leaving out `null` and `undefined` seems OK (we usually don't mention them in our error messages).

Reply:
Dictionary is definitely the correct term; we use it in undici's webidl quite a bit. https://github.com/nodejs/undici/blob/1a3707887c7366403a30a45e1d8335945d0f82c1/lib/fetch/webidl.js#L337

Reply:
Dictionary also sounds good to me, if that's generally understood. 👍